Setting pyglets 'audio' option fails

**Describe the bug**
The type check for setting the audio option, causes an exception, even when a tuple with strings is passed.

**System info**

Apple M1
OS: 14.5
Python: 3.9.18

**How To Reproduce**

```
pyglet.options["audio"] = ("openal", "xaudio2", "directsound", "pulse", "silent")
```

Causes

```
File "/Users/****/arcade/.venv/lib/python3.9/site-packages/pyglet/__init__.py", line 260, in __setitem__
    assert type(value).__name__ == self.__annotations__[key], f"Invalid type: '{type(value)}' for '{key}'"
AssertionError: Invalid type: '<class 'tuple'>' for 'audio'
```

The type check is too aggressive, the env variable `PYGLET_AUDIO` does not work anymore as well:
```
> PYGLET_AUDIO="openal,silent" python3 -m examples.media.soundspace.soundspace
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/square/Code/python/pyglet/examples/media/soundspace/soundspace.py", line 39, in <module>
    import pyglet
  File "/home/square/Code/python/pyglet/pyglet/__init__.py", line 257, in <module>
    options[_key] = _value.split(",")
    ~~~~~~~^^^^^^
  File "/home/square/Code/python/pyglet/pyglet/__init__.py", line 245, in __setitem__
    assert type(value).__name__ == self.__annotations__[key], f"Invalid type: '{type(value)}' for '{key}'"
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Invalid type: '<class 'list'>' for 'audio'
```

--- a/pyglet/__init__.py
+++ b/pyglet/__init__.py
@@ -6,8 +6,9 @@ from __future__ import annotations
 
 import os
 import sys
+from collections.abc import ItemsView, Sequence
 from dataclasses import dataclass
-from typing import TYPE_CHECKING, Any, ItemsView
+from typing import TYPE_CHECKING, Any
 
 #: The release version
 version = "2.0.15"
@@ -33,11 +34,21 @@ if getattr(sys, "frozen", None):
     _enable_optimisations = True
 
 
+_SPECIAL_OPTION_VALIDATORS = {
+    "audio": lambda x: isinstance(x, Sequence),
+    "vsync": lambda x: x is None or isinstance(x, bool),
+}
+
+_OPTION_TYPE_VALIDATORS = {
+    "bool": lambda x: isinstance(x, bool),
+    "int": lambda x: isinstance(x, int),
+}
+
 @dataclass
 class Options:
     """Dataclass for global pyglet options."""
 
-    audio: tuple = ("xaudio2", "directsound", "openal", "pulse", "silent")
+    audio: Sequence[str] = ("xaudio2", "directsound", "openal", "pulse", "silent")
     """A :py:class:`~typing.Sequence` of valid audio modules names. They will
      be tried from first to last until either a driver loads or no entries
      remain. See :ref:`guide-audio-driver-order` for more information.
@@ -123,7 +134,7 @@ class Options:
 
      .. versionadded:: 1.1"""
 
-    vsync: bool = None
+    vsync: bool | None = None
     """If set, the `pyglet.window.Window.vsync` property is ignored, and
      this option overrides it (to either force vsync on or off).  If unset,
      or set to None, the `pyglet.window.Window.vsync` property behaves
@@ -250,7 +261,8 @@ class Options:
 
     def __setitem__(self, key: str, value: Any) -> None:
         assert key in self.__annotations__, f"Invalid option name: '{key}'"
-        assert type(value).__name__ == self.__annotations__[key], f"Invalid type: '{type(value)}' for '{key}'"
+        assert (_SPECIAL_OPTION_VALIDATORS.get(key, None) or _OPTION_TYPE_VALIDATORS[self.__annotations__[key]])(value), \
+            f"Invalid type: '{type(value)}' for '{key}'"
         self.__dict__[key] = value
 
 

